### ISSUE TITLE
MkdirAll is not respecting opts.FS

### ISSUE THREAD

**Issue by nianyush** [otiai10/copy#153]
https://github.com/otiai10/copy/blob/e40f4e6d0e0b8f66457f5eb0ab6cf7f782d0208e/copy.go#L104

**Comment by nianyush** [comment]
probably should use an alterantive MkdirAll that is compatible with fs.FS like https://github.com/twpayne/go-vfs/blob/master/mkdirall.go#L16

**Comment by otiai10** [comment]
Thank you for your comment. Could you please create a test case that demonstrates the failure you described?

**Comment by mafredri** [comment]
Not sure if this is the use-case @nianyush was referring to, but I added a PR with some tests that don't work to illustrate the problem.

**Comment by nianyush** [comment]
yes, i think your pr exactly describes what i was facing earlier @mafredri
